tests/docs/skip_some.rst

```diff
@@ -80,3 +80,73 @@ Code in doctest should run only if version condition is satisfied:
 .. doctest-requires:: pytest>=1.0 pytest>=2.0
 
     >>> import pytest
+
+
+Remote data block code sandwiched in block codes
+===============================================
+
+This code block should work just fine::
+
+    >>> 1 + 1
+    2
+
+This should be skipped when remote data is not requested otherwise the test should fail::
+
+.. doctest-remote-data::
+    >>> 1 + 3
+    2
+
+This code block should work just fine::
+
+    >>> 1 + 1
+    2
+
+
+Remote data followed by plain block code
+========================================
+
+This one should be skipped when remote data is not requested otherwise the test should fail::
+
+.. doctest-remote-data::
+    >>> 1 + 3
+    2
+
+This code block should work just fine::
+
+    >>> 1 + 1
```

Comment:
How would we know if this is actually silently skipped too, say, because it thinks this code is still somehow part of remote date block above? That is, how are we sure the remote data block really ended before this line?

Reply:
Yes, I'm not sure how we could do that, as currently all the narrative docs testings are done as one test, if it's stopped somewhere the rest of the file is not being picked up. But then we have a tracelog with a failure. Whenever it's skipped, we have no easy way to see that the rest of the file was run as expected. Maybe add a failing one to the very bottom of the file? But then it's a failing test, and here we cannot really have it as an xfail, afaik.

Reply:
Yes, perhaps that test case is too complicated to be tested this way. If this is tested in the other unit tests, then we don't have to do it here?

Reply:
oh, good point, there is indeed a better handle of these in the unittest content.

Reply:
@pllim - So if we run this file with `--remote-data`, then it of course stops with a fail on L98. As expected. So I'm not sure what to do with the file itself, maybe only include it in when `--remote-data` is not used, and test the working of the remote-data stuff separately, only in unit tests?

Reply:
Maybe put those tests in a dedicated file that can be run with and without `--remote-data` ?

Reply:
I'm not sure what was the goal of having all those blocks that should fail ? Since the first failing test will fail the file anyway, we cannot test that the other blocks that should fail actually fail. So I guess we could test the expected failures in the unit tests, and in this file test blocks that should either pass or be skipped ?